src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -286,7 +286,7 @@ public readonly void Dispose()
 
         // Track ALL of the recently re-committed persisted nodes. This is so that we don't accidentally remove
         // recommitted persisted nodes (which will not get re-persisted).
-        private readonly ConcurrentDictionary<HashAndTinyPathAndHash, long> _persistedLastSeen = new(CollectionExtensions.LockPartitions, 4 * 4096);
+        private readonly ConcurrentDictionary<HashAndTinyPathAndHash, long>? _persistedLastSeen;
```

Comment:
We don't have to touch any other usages?

Reply:
Lazily allocated a few more (like the DirtyTrieCache isn't needed etc)

Changed Change and NodeKey to structs